scripts/sync_api_docs.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+import json
+import os
+import sys
+from importlib import import_module
+from typing import Any, List
+
+PACKAGE_INFO_PATH = "docs/packages.json"
+PACKAGE_PAGE_PATH = "docs/packages"
+
+
+PACKAGE_DOC_TEMPLATE = """{title}
+{underscore}
+
+{docstring}
+
+.. currentmodule:: snorkel.{package_name}
+
+.. autosummary::
+   :toctree: _autosummary/{package_name}/
+   :nosignatures:
+
+   {members}
+"""
+
+
+def get_title_and_underscore(package_name: str) -> str:
+    title = f"Snorkel {package_name.capitalize()} Package"
+    underscore = "-" * len(title)
+    return title, underscore
+
+
+def get_package_members(package: Any) -> List[str]:
+    members = []
+    for name in dir(package):
+        if name.startswith("_"):
+            continue
+        obj = getattr(package, name)
+        if isinstance(obj, type) or callable(obj):
+            members.append(name)
+    return members
+
+
+def main(check: bool) -> None:
```

Comment:
some quick high-level comments about what this file is executing?

Reply:
code is mostly self-explaining; this is just for folks who want to quickly get a sense